### DISCUSSION TITLE
Ignore directory during build/publish

### DISCUSSION THREAD

**Discussion by mpaglia0** [getpelican/pelican#3211] [Q&A] [answered]
I need to manage a password protected directory on my website to be ignored during build/publish.
During the build process this directory does not exist in the Pelican content structure so - during publishing - the remote directory is cancelled.
This is OK for a normal use but I need Pelican will ignore my XXX directory so the remote one is NOT cancelled during publish.

I would like to manage a directory with content that is NOT generated by Pelican itself.
This directory will contain hundreds of files and I need to publish it with an FTP client remotely (but inside the structure created by Pelican).
Since this directory does not exist in the 'content' folder, it is not generated the matching one in 'output'.
During 'invoke publish' I ask Pelican to CANCEL folders/files that are no more existing, so my remote 'extra' folder is duly cancelled...

If not possible I can consider to create a soft link in the 'content' folder pointing to the folder containing my files and then add this soft link to STATIC_PATHS list in pelicanconf.py?

**Comment by justinmayer** [comment, marked as answer]
Based on what you have explained, I assume you are using the [default Rsync behavior of `invoke publish`](https://github.com/getpelican/pelican/blob/36b4c6dc0888a150220d9f5b1cc58107490ee4b0/pelican/tools/templates/tasks.py.jinja2#L158-L162), which does indeed delete all files and folders on the remote system that are not present on the local system. After doing some research about protecting files/folders from Rsync's delete behavior, I suggest that you edit the `publish` task in your `tasks.py` file and add the following to the `rsync` invocation:

    --filter 'protect /remote-directory-to-keep/'

**Reply by justinmayer** [reply]
I converted this to a discussion, moved the additional detail from comments to the original question, and provided an answer above.

If the above answer does not provide the behavior you are looking for, you can indeed try the other `STATIC_PATHS ` approach you mentioned. I do that with my talks on the [Justin Mayer](https://justinmayer.com) web site, which exist as a folder within my content folder (`content/talks/`). In your case, a symlink should achieve a similar effect. But I would try the suggested answer above first to see whether it meets your needs.

**Reply by mpaglia0** [reply]
The rsync parameter works perfectly as desired!
Thank you and have a nice day,
Maurizio
